modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStoreFactory.java

```diff
@@ -146,7 +149,20 @@
             "SQL statement executed when the connection is released to the pool", false, null,
             Collections.singletonMap(Parameter.IS_LARGE_TEXT, Boolean.TRUE));
 
-    
+    /** JDBC callback factory */
+    public static final Param CALLBACK_FACTORY = new Param("Callback factory", String.class, 
```

Comment:
While it is certainly an advanced parameter it's not just intended for programatic use, it's meant to be configured by the user as well.